models/User.js

```diff
@@ -14,6 +14,25 @@ let UserSchema = new Schema({
     name:           { first: String, last: String }
 });
 
+// Temp User model for email verification. 
+// Everything is the same as persistent User model, 
+// except generated URL and enrollment codes entered at registration.
+let TempUserSchema = new Schema({
+    _id: { type: String },
+    email:          { type: String, required: true },
+    password:       { type: String, required: true },                 // Hashed.
+    instructor:     { type: Boolean, required: true },                
+    admin:          { type: Boolean, default: false, required: true }, 
+    name:           { first: String, last: String },
+    GENERATED_VERIFYING_URL: {type: String, required: true},
+    enroll_codes: {type: [String], required: false},
+    createdAt: {  // TTL of tempuser
+        type: Date,
+        expireAfterSeconds : 86400,
+        default: Date.now
+    },
+});
+
```

Comment:
Can't you just have a boolean "verification" attribute on the basic user? 
verification: { isVerified: Boolean, verificationURL: String }

Your solution seems needlessly complex, but maybe you've already considered the option I suggested.

Reply:
Okay I just considered that my solution would involve checking for "isVerified" on every single User query we do. Is there a way to circumvent this problem using middleware without involving so much duplicated code and added complexity?

Reply:
I will look into mongoose inheritance instead.
For documentation's sake:
I think the boolean isn't a good idea because
1) it takes up extra space for a flag that's essentially used once
2) doesn't account for case where user is never verified, so they will sit in persistent DB forever (we wouldn't be able to use mongo's TTL deletion and would have to add complex logic to implement it ourselves which is undesirable)

I will also double check that mongo's TTL script is actually running. I'm not sure why it wasn't deleting things when my code was technically correct, and it WAS aware that there was a TTL index.